src/detectors/signal_based.hpp

```diff
@@ -31,26 +30,25 @@ namespace mesos {
 namespace serenity {
 
 /**
- * DetectorFilter is able to check defined value and trigger some contentions
+ * SignalBasedDetector is able to check defined value and trigger some contentions
  * on given thresholds.
  */
-class ContentionDetectorFilter :
+class SignalBasedDetector :
     public Consumer<ResourceUsage>,
     public Producer<Contentions> {
  public:
-  ContentionDetectorFilter(
+  SignalBasedDetector(
       Consumer<Contentions>* _consumer,
       const lambda::function<usage::GetterFunction>& _valueGetFunction,
       SerenityConfig _detectorConf,
       const Tag& _tag = Tag(QOS_CONTROLLER, "detectorFilter"))
```

Comment:
I would propose a separate term... We have observers (usage -> action), filters (usage->usage), detectors. (usage->contention). Do you agree? (:.